### PR TITLE
CI using artifact caching proxy

### PR DESCRIPTION
**Pull request by alecharp**
### Description

Trying to use artifact caching proxy to build the project. This should speed up the build a tiny bit.

### Submitter checklist

- [x] If the issue exists, it is well described and linked in the description
- [x] The description of this pull request is detailed and explain why this pull request is needed
- [x] The changeset is on a specific branch
  - `feature/` for new feature, or improvements
  - `fix/` for bug fixes
  - `docs/` for any documentation changes
- [x] If required, the documentation has been updated
- [x] There is automated tests to cover the code change / addition
    - If there is no test, include an explanation why in the description
- [x] Run `mvn verify` locally and all tests are passing successfully
  - It is OK to create a pull request which has failing tests if it is created as a draft, is to fix a bug and the first commit is the test to prove the existence of the bug.
- [x] There is no new warnings (checkstyle nor spotbugs) on the code

### CODE REVIEW

**Comment by alecharp** [comment]
Just by comparing last `build` stage in `main` branch and the same stage in this pull request build, we are saving 20min of build time.

File: Jenkinsfile [hunk 1]
```diff
@@ -17,7 +17,24 @@ pipeline {
         JAVA_HOME = '/opt/jdk-17/'
       }
       steps {
-        sh './mvnw -V verify checkstyle:checkstyle spotbugs:spotbugs -Dmaven.test.failure.ignore -Dcheckstyle.failOnViolation=false -Dspotbugs.failOnError=false'
+        script {
+          def OPTS=''
+          if(env.MAVEN_SETTINGS) {
+            OPTS="-s ${MAVEN_SETTINGS}"
+          }
+          infra.withArtifactCachingProxy() {
```
[the remark this answers is not in the document]
Comment (alecharp) [reply Jenkinsfile:25]: I keep forgetting about this syntax. Thanks!

**Comment by alecharp** [comment]
After re-reading https://groovy-lang.org/operators.html#_elvis_operator, the operator is not what we wanted exactly.

File: Jenkinsfile [hunk 2]
```diff
@@ -17,7 +17,21 @@ pipeline {
         JAVA_HOME = '/opt/jdk-17/'
       }
       steps {
-        sh './mvnw -V verify checkstyle:checkstyle spotbugs:spotbugs -Dmaven.test.failure.ignore -Dcheckstyle.failOnViolation=false -Dspotbugs.failOnError=false'
+        script {
+          infra.withArtifactCachingProxy() {
+            def OPTS = env.MAVEN_SETTINGS ? "-s ${MAVEN_SETTINGS}" : ''
+            sh """
+              ./mvnw -V \
+                ${OPTS} \
```
Comment (lemeurherve) [inline Jenkinsfile:25]:
```suggestion
                --no-transfer-progress \
                ${OPTS} \
```
Nit, for shorter logs

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**